__tests__/app.test.js

```diff
@@ -95,37 +95,99 @@ describe("/api/articles", () => {
+						expect(typeof comment.votes).toBe("number");
+						expect(typeof comment.created_at).toBe("string");
+						expect(typeof comment.author).toBe("string");
+						expect(typeof comment.body).toBe("string");
+						expect(comment.article_id).toBe(1);
+					});
 				});
-			});
-	});
-	test("returns 400 and bad request when invalid id type requested", () => {
-		return request(app)
-			.get("/api/articles/hello/comments")
-			.expect(400)
-			.then(({ body }) => {
-				expect(body.msg).toBe("Bad request");
-			});
+		});
+		test("GET returns 400 and bad request when invalid id type requested", () => {
+			return request(app)
+				.get("/api/articles/hello/comments")
+				.expect(400)
+				.then(({ body }) => {
+					expect(body.msg).toBe("Bad request");
+				});
+		});
+		test("GET returns 404 and author id does not exist when non-existant author id requested", () => {
+			return request(app)
+				.get("/api/articles/600000000/comments")
+				.expect(404)
+				.then(({ body }) => {
+					expect(body.msg).toBe("Author id does not exist");
+				});
+		});
 	});
-	test("returns 404 and author id does not exist when non-existant author id requested", () => {
-		return request(app)
-			.get("/api/articles/600000000/comments")
-			.expect(404)
-			.then(({ body }) => {
-				expect(body.msg).toBe("Author id does not exist");
-			});
+	describe("POST", () => {
+		test("responds 201 and posts new comment", () => {
+			const comment = {
+				username: "rogersop",
+				body: "loved this",
+			};
+			return request(app)
+				.post("/api/articles/1/comments")
+				.send(comment)
+				.expect(201)
+				.then(({ body }) => {
+					expect(body.comment.comment_id).toBe(19);
+					expect(body.comment.body).toBe("loved this");
+					expect(body.comment.article_id).toBe(1);
+					expect(body.comment.author).toBe("rogersop");
+					expect(body.comment.votes).toBe(0);
+					expect(typeof body.comment.created_at).toBe("string");
+				});
+		});
+		test("responds 400 missing username or body", () => {
+			const comment = {
+				wrong: "nope",
+			};
+			return request(app)
+				.post("/api/articles/1/comments")
+				.send(comment)
+				.expect(400)
+				.then(({ body }) => {
+					expect(body.msg).toBe("missing either username or body");
+				});
+		});
+		test("responds 400 bad request when article_id invalid", () => {
+			const comment = {
+				username: "rogersop",
+				body: "loved this",
+			};
+			return request(app)
+				.post("/api/articles/hello/comments")
+				.send(comment)
+				.expect(400)
+				.then(({ body }) => {
+					expect(body.msg).toBe("Bad request");
+				});
+		});
+		test("responds 404 article not found when article_id nonexistant", () => {
+			const comment = {
+				username: "rogersop",
+				body: "loved this",
+			};
+			return request(app)
+				.post("/api/articles/50000/comments")
+				.send(comment)
+				.expect(404)
+				.then(({ body }) => {
+					expect(body.msg).toBe("article not found");
+				});
```

Comment:
one more test for non-existent user would be good!

Reply:
added this now and added a new pr! thanks